photon-ml/src/test/scala/com/linkedin/photon/ml/function/ObjectiveFunctionTest.scala

```diff
@@ -16,343 +16,501 @@ package com.linkedin.photon.ml.function
 
 import java.util.Random
 
-import scala.collection.mutable
-
 import breeze.linalg.{DenseVector, SparseVector, Vector}
 import com.linkedin.photon.ml.data.LabeledPoint
-import com.linkedin.photon.ml.optimization._
+import com.linkedin.photon.ml.supervised.TaskType
 import com.linkedin.photon.ml.test.SparkTestUtils
 import org.apache.log4j.{LogManager, Logger}
 import org.testng.Assert.{assertEquals, assertTrue}
 import org.testng.annotations.{DataProvider, Test}
 
-
 /**
- * Unit tests to check to verify the gradients / Hessians and other methods
- * @author bdrew
- * @author yali
- */
+  * Unit tests to verify that the loss functions compute gradients & Hessians accurately.
+  */
 class ObjectiveFunctionTest extends SparkTestUtils {
```

Comment:
@joshvfleming 
Needs to be renamed, now that there is an actual ObjectiveFunction class. TBH, this class is more of an integration test than a unit test.